Tests/SonarScanner.MSBuild.Tasks.UnitTests/GetAnalyzerSettingsTests.cs

```diff
@@ -305,144 +273,140 @@ public void GetAnalyzerSettings_ConfigExists_NewBehaviour_SettingsMerged()
-
-            // Assert
-            executedTask.RuleSetFilePath.Should().Be("c:\\test.project.ruleset");
-            executedTask.AnalyzerFilePaths.Should().BeEquivalentTo("c:\\sonar.csharp1.dll", "c:\\Google.Protobuf.dll");
-            executedTask.AdditionalFilePaths.Should().BeEquivalentTo("c:\\add1.txt", "d:\\replaced1.txt");
-        }
-
-        [TestMethod]
-        public void GetAnalyzerSettings_ConfigExists_NewBehaviour_TestProject_SonarAnalyzerSettingsUsed_CSharp()
+        [DataTestMethod]
+        [DataRow("7.3", "cs", @"c:\csharp-normal.ruleset", DisplayName = "Legacy CS")]
+        [DataRow("7.4", "cs", @"c:\csharp-normal.ruleset")]
+        [DataRow("7.3", "vbnet", @"c:\vbnet-normal.ruleset", DisplayName = "Legacy VB")]
+        [DataRow("7.4", "vbnet", @"c:\vbnet-normal.ruleset")]
+        public void GetAnalyzerSettings_ConfigExists_ProductProject_SonarAnalyzerSettingsUsed(string sonarQubeVersion, string language, string expectedRuleset)
         {
             // Arrange and Act
-            var executedTask = ExecuteGetAnalyzerSettings_ConfigExists_TestProject_SonarAnalyzerSettingsUsed("7.4", "cs");
+            var executedTask = ExecuteGetAnalyzerSettings_ConfigExists_TestProject_SonarAnalyzerSettingsUsed(sonarQubeVersion, language, false, null);
 
             // Assert
-            executedTask.RuleSetFilePath.Should().Be("c:\\test.project.ruleset");
-            executedTask.AnalyzerFilePaths.Should().BeEquivalentTo("c:\\sonar.csharp1.dll", "c:\\Google.Protobuf.dll");
-            executedTask.AdditionalFilePaths.Should().BeEquivalentTo("c:\\add1.txt", "d:\\replaced1.txt");
+            executedTask.RuleSetFilePath.Should().Be(expectedRuleset);
+            executedTask.AnalyzerFilePaths.Should().BeEquivalentTo(@"c:\wintellect1.dll", @"c:\Google.Protobuf.dll", $@"c:\sonar.{language}.dll", @"c:\Google.Protobuf.dll");
+            executedTask.AdditionalFilePaths.Should().BeEquivalentTo($@"c:\add1.{language}.txt", @"d:\replaced1.txt", "original.should.be.preserved.for.product.txt");
         }
 
-        [TestMethod]
-        public void GetAnalyzerSettings_ConfigExists_Legacy_TestProject_SonarAnalyzerSettingsUsed_VB()
+        [DataTestMethod]
+        [DataRow("7.3", "cs", @"c:\csharp-normal.ruleset", /* not set */ null, DisplayName = "Legacy CS")]
+        [DataRow("7.4", "cs", @"c:\csharp-normal.ruleset", "false")]
+        [DataRow("7.4", "cs", @"c:\csharp-normal.ruleset", "FALSE")]
+        [DataRow("7.4", "cs", @"c:\csharp-normal.ruleset", "UnexpectedParamValue")]
+        [DataRow("7.3", "vbnet", @"c:\vbnet-normal.ruleset", /* not set */ null, DisplayName = "Legacy VB")]
+        [DataRow("7.4", "vbnet", @"c:\vbnet-normal.ruleset", /* not set */ null)]
+        public void GetAnalyzerSettings_ConfigExists_TestProject_SonarAnalyzerSettingsUsed(string sonarQubeVersion, string language, string expectedRuleset, string excludeTestProject)
```

Comment:
the name of the test should contain the fact that test projects are not excluded (otherwise, I ask myself why only `false` or `/* not set */` are used, and not also `True`).

I see two test names:
- GetAnalyzerSettings_ConfigExists_TestProject_SonarAnalyzerSettingsUsed
- GetAnalyzerSettings_ConfigExists_ExcludedTestProject_DeactivatedSonarAnalyzerSettingsUsed

Both should contain that the project is a TestProject (`true` used inside), like the first name does, and one should say something like "TestsNotExcluded" and "TestsExcluded", to clarify what the `excludeTestProject` parameter should be in the DataRows.

And could you replace the `GetAnalyzerSettings` with something shorter like `Execute`? Because the task name is in the test class name.

We'd end up with 
- Execute_ForTestProject_ConfigExists_TestsNotExcluded_SonarAnalyzerSettingsUsed
- Execute_ForTestProject_ConfigExists_TestsExcluded_DeactivatedSonarAnalyzerSettingsUsed

Reply:
I've deleted `GetAnalyzerSettings_` from all names. It should not be there at all.
I've ended up with:
```
ConfigExists_ForProductProject_SonarAnalyzerSettingsUsed
ConfigExists_ForTestProject_WhenAnalyzed_SonarAnalyzerSettingsUsed
ConfigExists_ForTestProject_WhenExcluded_DeactivatedSonarAnalyzerSettingsUsed
ConfigExists_ForTestProject_WhenUnknownLanguage_NewBehaviour_SonarAnalyzerSettingsUsed
```

Reply:
I don't understand what "NewBehaviour" means, and I spent some time reading the code. 

Could you remove it or find a better name, or detail it in a comment?

Reply:
As the `new` means `current`, I'll just remove it. It doesn't make sense to name stuff as `new` in the long run. (Yes, I'm looking at you, `New York` and [Nové Město](https://en.mapy.cz/zakladni?x=15.7418352&y=49.7796139&z=8&q=nov%C3%A9%20m%C4%9Bsto) = `New Town`)